src/beast/beast/net/IPAddress.h

```diff
@@ -142,12 +142,12 @@ class Address
     template <class Hasher>
     friend
     void
-    hash_append(Hasher& h, Address const& addr)
+    hash_append(Hasher& h, Address const& addr) noexcept
     {
         using beast::hash_append;
         if (addr.is_v4 ())
             hash_append(h, addr.to_v4 ());
-        else
+        else if (addr.is_v6 ())
             hash_append(h, addr.to_v6 ());
```

Comment:
What if it's neither? When the proverbial IP v7 does come along, I'd hate for this function to potentially get missed and become a NOOP.